File: storeGraphInList.py
def withOutWeight():
    nodes = int(input("Enter No of nodes: "))
    edges = int(input("Enter No of edges: "))

    adj = [[] for _ in range(nodes + 1)]

    for i in range(edges):
        u = int(input("Enter node: "))
        v = int(input("Enter connected node: "))

        adj[u].append(v)
        adj[v].append(u)

    print("Adjacency List:")
    for i in range(1, nodes + 1):
        print(f"{i} -> {' '.join(map(str, adj[i]))}")

def withWeight():
    nodes = int(input("Enter No of nodes: "))
    edges = int(input("Enter No of edges: "))

    graph = {}

    for i in range(edges):
        u = int(input("Enter node: "))
        v = int(input("Enter connected node: "))
        w = int(input("Enter connected weight: "))

        if u not in graph:
            graph[u] = {}
        if v not in graph:
            graph[v] = {}

        graph[u][v] = w
        graph[v][u] = w

    print("Adjacency List with Weights:")
    for u, edges in graph.items():
        print(f"{u} -> {' '.join([f'({v}, {w})' for v, w in edges.items()])}")

if __name__ == "__main__":
    # Uncomment one of the following to run the desired function
    # withOutWeight()
    withWeight()

File: storeGraphInList.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

void withOutWeight() {
    cout << "Enter No of nodes: ";
    int nodes;
    cin >> nodes;

    cout << "Enter No of edges: ";
    int edges;
    cin >> edges;

    vector<vector<int>> adj(nodes + 1);

    for (int i = 0; i <= nodes; i++) {
        adj[i] = vector<int>();
    }

    for (int i = 0; i < edges; i++) {
        cout << "Enter node: ";
        int u, v;
        cin >> u;

        cout << "Enter connected node: ";
        cin >> v;

        adj[u].push_back(v);
        adj[v].push_back(u);
    }

    cout << "Adjacency List: " << endl;
    for (int i = 1; i <= nodes; i++) {
        cout << i << " -> ";
        for (int j : adj[i]) {
            cout << j << " ";
        }
        cout << endl;
    }
}

void withWeight() {
    cout << "Enter No of nodes: ";
    int nodes;
    cin >> nodes;

    cout << "Enter No of edges: ";
    int edges;
    cin >> edges;

    map<int, map<int, int>> graph;

    for (int i = 0; i < edges; i++) {
        cout << "Enter node: ";
        int u, v;
        cin >> u;

        cout << "Enter connected node: ";
        cin >> v;

        cout << "Enter connected weight: ";
        int w;
        cin >> w;

        graph[u][v] = w;
        graph[v][u] = w;
    }

    cout << "Adjacency List with Weights: " << endl;
    for (const auto& entry : graph) {
        int u = entry.first;
        const auto& edges = entry.second;
        cout << u << " -> ";
        for (const auto& edge : edges) {
            int v = edge.first;
            int w = edge.second;
            cout << "(" << v << ", " << w << ") ";
        }
        cout << endl;
    }
}

int main() {
    // Uncomment one of the following to run the desired function
    // withOutWeight();
    withWeight();
    return 0;
}
